File: src/main/java/com/integreety/yatspec/e2e/captor/http/mapper/PropertyServiceNameDeriver.java
package com.integreety.yatspec.e2e.captor.http.mapper;

public class PropertyServiceNameDeriver {

    private static final String DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE = " Service";

    private final String appName;

    public PropertyServiceNameDeriver(final String appName) {
        this.appName = appName
                .replaceAll(DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE, "")
                .replaceAll(" ", "");
    }

    public String getServiceName() {
        return appName;
    }
}

File: core/src/main/kotlin/io/lsdconsulting/lsd/distributed/interceptor/captor/common/PropertyServiceNameDeriver.kt
package io.lsdconsulting.lsd.distributed.interceptor.captor.common

class PropertyServiceNameDeriver(appName: String) {
    val serviceName: String = appName
        .replace(DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE.toRegex(), "")

    companion object {
        // TODO This should be configurable
        private const val DEFAULT_SERVICE_NAME_SUFFIX_TO_REMOVE = " Service"
    }
}
